frontend/lib/features/home/home_page.dart

```diff
@@ -1,21 +1,169 @@
 import 'package:flutter/material.dart';
-import 'package:flutter_starter/features/home/account_balance.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_starter/features/deposit/deposit_page.dart';
+import 'package:flutter_starter/features/withdraw/withdraw_page.dart';
 import 'package:flutter_starter/l10n/app_localizations.dart';
-import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:flutter_starter/shared/grid.dart';
 
-class HomePage extends HookConsumerWidget {
-  const HomePage({super.key});
+class HomePage extends HookWidget {
+  HomePage({super.key});
+
+  final List<Transaction> txns = [];
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
+  Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(title: Text(Loc.of(context).home)),
-      body: const Column(
-        crossAxisAlignment: CrossAxisAlignment.stretch,
+      body: SafeArea(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            buildAccountBalance(context),
+            Padding(
+              padding: const EdgeInsets.symmetric(
+                  horizontal: Grid.side, vertical: Grid.xs),
+              child: Text(
+                Loc.of(context).activity,
+                style: Theme.of(context).textTheme.bodyMedium,
+              ),
+            ),
+            Expanded(
+              child: txns.isEmpty
+                  ? buildEmptyState(context)
+                  : ListView(
```

Comment:
We could move this to a func to make this scaffolding really easy to parse quickly